l10n_ch_invoice_reports/models/report.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import io
+from io import BytesIO
+
+import PyPDF2
+
+from odoo import models
+
+
+class IrActionsReport(models.Model):
+    _inherit = "ir.actions.report"
+
+    def merge_pdf_in_memory(self, docs):
+        streams = []
+        writer = PyPDF2.PdfFileWriter()
+        for doc in docs:
+            if doc:
+                current_buff = BytesIO()
+                streams.append(current_buff)
+                current_buff.write(doc.read())
+                current_buff.seek(0)
+                reader = PyPDF2.PdfFileReader(current_buff)
+                for page in range(reader.getNumPages()):
+                    writer.addPage(reader.getPage(page))
+            else:
+                writer.addBlankPage()
+        buff = BytesIO()
+        try:
+            # The writer close the reader file here
+            writer.write(buff)
+            return buff.getvalue()
+        except IOError:
+            raise
+        finally:
+            buff.close()
+            for stream in streams:
+                stream.close()
+
+    def render_qweb_pdf(self, res_ids=None, data=None):
+        reports = [
+            "l10n_ch_invoice_reports.invoice_isr_report_main",
+            "l10n_ch_invoice_reports.invoice_qr_report_main",
+            "l10n_ch_invoice_reports.invoice_qr_isr_report_main",
+        ]
+        ctx = self.env.context
+        # Remove context key that comes from invoice action
+        if ctx.get("default_type"):
+            ctx = ctx.copy()
+            ctx.pop("default_type")
+        if self.report_name not in reports or not res_ids:
+            return super().render_qweb_pdf(res_ids, data)
+        inv_report = self._get_report_from_name("account.report_invoice")
+        invoice_pdf, _ = inv_report.render_qweb_pdf(res_ids, data)
+        invoice_pdf_io = io.BytesIO(invoice_pdf)
+
+        isr_report = self._get_report_from_name("l10n_ch.isr_report_main")
+        isr_pdf, _ = isr_report.render_qweb_pdf(res_ids, data)
+        isr_pdf_io = io.BytesIO(isr_pdf)
+
+        qr_report = self._get_report_from_name("l10n_ch.qr_report_main")
+        qr_pdf, _ = qr_report.render_qweb_pdf(res_ids, data)
+        qr_pdf_io = io.BytesIO(qr_pdf)
```

Comment:
no need to close these objects?

Reply:
They are closed a few lines down https://github.com/OCA/l10n-switzerland/pull/581/files#diff-95e065a95f81606f9d2f3111e3db2866670a00966c9e8442d6168e756046a06eR67